### PR TITLE
LESQ-1045: implement filter drawer on unit pages

### PR DESCRIPTION
**Pull request by k-huggs**
## Description

Music year: 2022

Implements filter drawer component on specialist and non-specialist unit listing pages

- Creates `MobileUnitFilters` component and test file
- Renders filter drawer on Specialist and Non-Specialist unit listing pages 

## Acceptance Criteria:

- [x]  Make the search filter drawer and the unit listing drawer the same component
Filter drawer component created here: https://github.com/oaknational/oak-components/pull/291 not implemented on Search as of yet 
- [x]  Close, clear and (sticky) show results buttons
- [x]  Dropdown shouldn’t close until a user interacts with the close or show results buttons
    - Close cancels the action and doesn’t apply the selected filters, closes the dropdown
    - Show results applies the selected filters, closes the dropdown
- [x]  Number of results should be on the apply button
- [x]  Style!

## Issue(s)

Fixes #

## How to test

1. Go to https://deploy-preview-2835--oak-web-application.netlify.thenational.academy
https://deploy-preview-2835--oak-web-application.netlify.thenational.academy/teachers/programmes/combined-science-secondary-ks4-foundation-edexcel/units
2. Navigate to any unit page with filters
3. You should see on mobile and tablet view filter button is rendered, when selecting this, the FilterDrawer should open
4. When clicking the filters, selected filter is highlighted, the draw doesn't close, the submit button is update with number of units
5. Clear button clears all the filters, X button closes the filter drawer without applying changes, submit button updates query and correct units displayed.

## Screenshots

How it should now look:
(Re-uploading svg on filter button)

https://github.com/user-attachments/assets/4149b8c6-34db-4dd3-8043-f884104d8f16

## Checklist

- [ ] Added or updated tests where appropriate
- [ ] Manually tested across browsers / devices
- [ ] Considered impact on accessibility
- [ ] Design sign-off
- [ ] Approved by product owner
- [ ] Does this PR update a package with a breaking change

### CODE REVIEW

**Review by jack-skerrett-oak — CHANGES_REQUESTED** [review]
- svg icon isn’t rendering on the filter drawer button
- There is a rule in figma that states “On mobile, the drawer spans the full width of the screen.”, currently this implementation only fills the whole screen when the filter drawer width is the width of the screen, it doesn’t pay attention to the breakpoints

**Review by Monika-Koziol — APPROVED** [review]
Jack has already raised the icon problem. All good otherwise.

**Review by jack-skerrett-oak — APPROVED** [review]

File: src/components/TeacherComponents/UnitsLearningThemeFilters/UnitsLearningThemeFilters.tsx [hunk 1]
```diff
@@ -58,47 +64,51 @@ const UnitsLearningThemeFilters = ({
         })
     : [];
   const router = useRouter();
+  const isMobile = idSuffix === "mobile";
 
   const categorySlug = router.query["category"]?.toString();
   const yearGroupSlug = router.query["year"]?.toString();
 
-  const { track } = useAnalytics();
```
[the remark this answers is not in the document]
Comment (k-huggs) [reply src/components/TeacherComponents/UnitsLearningThemeFilters/UnitsLearningThemeFilters.tsx:65]: Wanted to decouple the analytics provider from the component, we had some issues previously with testability and flexibility/reusability when each component was in charge of their own analytics. So we moved the providers to the view level and passed the functions in

File: src/components/TeacherComponents/MobileUnitFilters/MobileUnitFilters.tsx [hunk 2]
```diff
@@ -0,0 +1,225 @@
...
+  numberOfUnits: number;
+  browseRefined: TrackFns["browseRefined"];
+  setSelectedThemeSlug: (theme: string | undefined) => void;
+  learningThemesFilterId: string;
+  isSpecialist?: boolean;
+} & (UnitListingData | SpecialistUnitListingData);
+
+// Type guard as due to differences in specialist and non-specialist unit listing data
+function isUnitListingData(
+  data: UnitListingData | SpecialistUnitListingData,
```
[the remark this answers is not in the document]
Comment (k-huggs) [reply src/components/TeacherComponents/MobileUnitFilters/MobileUnitFilters.tsx:32]: Agreed, differences are quite slight but make things extra tricky

File: src/components/TeacherComponents/MobileUnitFilters/MobileUnitFilters.tsx [hunk 3]
```diff
@@ -0,0 +1,225 @@
...
+    setIsOpen(false);
+  };
+
+  const handleClose = () => {
+    setIsOpen(false);
+    setYear(router.query["year"]?.toString() ?? "");
+    setCategory(router.query["category"]?.toString() ?? "");
+    setTheme(router.query["learning-theme"]?.toString() ?? "");
+  };
+  // Cloudinary svg system icon not accepting color props
```
[the remark this answers is not in the document]
Comment (k-huggs) [reply src/components/TeacherComponents/MobileUnitFilters/MobileUnitFilters.tsx:134]: This was an old comment to myself, the previous SVG I uploaded had fill set to a specific colour which was why it was broken

File: src/components/TeacherComponents/SubjectCategoryFilters/SubjectCategoryFilters.tsx [hunk 4]
```diff
@@ -65,81 +71,93 @@ const SubjectCategoryFilters: React.FC<SubjectCategoryFiltersProps> = ({
...
-                      ...(selectedThemeSlug && {
-                        "learning-theme": selectedThemeSlug,
-                      }),
-                      programmeSlug,
-                      category: category.slug,
-                      ...(router.query.year && {
-                        year: router.query.year,
-                      }),
+              } else {
+                setCategory && setCategory("");
```
[the remark this answers is not in the document]
Comment (k-huggs) [reply src/components/TeacherComponents/SubjectCategoryFilters/SubjectCategoryFilters.tsx]: Changes pushed up

File: src/components/TeacherComponents/UnitsLearningThemeFilters/UnitsLearningThemeFilters.tsx (hunk 1 above)
[the remark this answers is not in the document]
Comment (carlmw) [reply src/components/TeacherComponents/UnitsLearningThemeFilters/UnitsLearningThemeFilters.tsx:65]: So long as this won't cause any issues with timing/initialisation I guess that is ok.

**Review by carlmw — APPROVED** [review]

**Pull request merged**